File: aoc2023/day05/Day5.cs

public class Day5
{
    public static void Run()
    {
        using (StreamReader file = new StreamReader("day05/p.in"))
        {
            string? ln;
            while ((ln = file.ReadLine()) != null)
            {
                Console.WriteLine(ln);
            }

            file.Close();
        }
    }
}

File: aoc2019/day05/Day5.fs
module Day5
open System.IO
open System

type Day5() =
    static member Run() =
        use file = new StreamReader("day05/p.in")

        let rec readLines() =
            match file.ReadLine() with
            | null -> ()
            | ln ->
                Console.WriteLine(ln)
                readLines()

        readLines()
